src/test/java/bio/terra/app/controller/AdminApiControllerTest.java

```diff
@@ -0,0 +1,155 @@
+import bio.terra.service.dataset.exception.DatasetNotFoundException;
+import bio.terra.service.filedata.DrsService;
+import bio.terra.service.job.JobService;
+import bio.terra.service.snapshot.SnapshotService;
+import bio.terra.service.snapshot.exception.SnapshotNotFoundException;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+@ActiveProfiles({"google", "unittest"})
+@ContextConfiguration(classes = {AdminApiController.class, GlobalExceptionHandler.class})
+@Tag(Unit.TAG)
+@WebMvcTest
+class AdminApiControllerTest {
+
+  @Autowired private MockMvc mvc;
+
+  @MockBean private JobService jobService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private DrsService drsService;
+  @MockBean private IamService iamService;
+  @MockBean private DatasetService datasetService;
+  @MockBean private SnapshotService snapshotService;
+  @MockBean private ApplicationConfiguration applicationConfiguration;
+
+  private static final AuthenticatedUserRequest TEST_USER =
+      AuthenticationFixtures.randomUserRequest();
+  private static final UUID MODEL_ID = UUID.randomUUID();
+  private static final String ADMIN_DATASETS_ENDPOINT = "/api/admin/v1/datasets/{id}";
+  private static final String ADMIN_SNAPSHOTS_ENDPOINT = "/api/admin/v1/snapshots/{id}";
+  private static final IamForbiddenException FORBIDDEN_EXCEPTION =
+      new IamForbiddenException("Forbidden");
+
+  @BeforeEach
+  void setUp() {
+    when(authenticatedUserRequestFactory.from(any())).thenReturn(TEST_USER);
+  }
+
+  @Test
+  void testAdminRetrieveDataset() throws Exception {
+    when(iamService.isResourceTypeAdminAuthorized(any(), any(), any())).thenReturn(true);
+    when(datasetService.retrieveDatasetModel(any(), any(), any()))
+        .thenReturn(new DatasetModel().id(MODEL_ID));
+    String json =
+        mvc.perform(get(ADMIN_DATASETS_ENDPOINT, MODEL_ID))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    DatasetModel model = TestUtils.mapFromJson(json, DatasetModel.class);
+    assertThat("Dataset Model is returned", model, notNullValue());
+    assertThat("Dataset Model ID is correct", model.getId(), equalTo(MODEL_ID));
+  }
+
+  @Test
+  void testAdminRetrieveDatasetInvalidId() throws Exception {
+    mvc.perform(get(ADMIN_DATASETS_ENDPOINT, "not a UUID"))
+        .andExpect(status().isBadRequest())
+        .andReturn()
+        .getResponse()
+        .getStatus();
```

Comment:
If you don't need the result, I think you can omit everything after `andExpect()`